### ISSUE TITLE
Selector '> ul > li p'

### ISSUE THREAD

**Issue by maximus8891** [tailwindlabs/tailwindcss-typography#266]
### What version of @tailwindcss/typography are you using?

v0.5.2

### What version of Node.js are you using?

v17.8.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### Reproduction repository

https://codesandbox.io/s/immutable-sound-2lzxpr?file=/index.html

### Describe your issue

The config line for `> ul > li p` is transpiled to `.prose > :where(ul > li p):not(:where([class~='not-prose'] *))` as per [the test](https://github.com/tailwindlabs/tailwindcss-typography/blob/master/src/index.test.js#L161). The issue is that the transpiled line does not do what I expected it to do as demonstrated in the [codesandbox](https://codesandbox.io/s/immutable-sound-2lzxpr?file=/index.html).

Selector `.prose > :where(ul > li p)` is equivalent to `ul > li .prose > p` and not `.prose > ul > li p`. Was it the intention?

**Comment by adamwathan** [comment]
Good catch! I think we need to change all of this stuff to include `.prose` in the `:where`, like:

```
.prose :where(.prose > ul > li p)
```

Would welcome a PR that tries to sort that out if interested, otherwise will hopefully get to it in the next few weeks 👍
